src/main/java/com/iexec/core/security/JwtTokenProvider.java

```diff
@@ -108,7 +99,7 @@ public String getWalletAddress(String token) {
 
     public String getWalletAddressFromBearerToken(String bearerToken) {
         String token = resolveToken(bearerToken);
-        if (token != null && isValidToken(token)) {
+        if (token != null) {
             return getWalletAddress(token);
```

Comment:
Wouldn't `getWalletAddress(token)` fail if the token is not valid.

Reply:
it depends on how the token is not valid, it can throw a runtime exception or just return the value stored in the `audience` field.
When retrieving the challenge and verifying the signature, we will know if we retrieved a valid address or not.